test/dialect/mysql/test_types.py

```diff
@@ -478,6 +479,34 @@ def test_float_type_compile(self, type_, sql_text):
         self.assert_compile(type_, sql_text)
 
 
+class TypeCompileUUIDMariaDBTest(fixtures.TestBase, AssertsCompiledSQL):
+    __only_on__ = "mariadb"
+
+    @testing.only_if("mariadb>=10.7")
+    def test_uuid(self):
+        self.assert_compile(sqltypes.Uuid(), "UUID")
+
+    @testing.only_if("mariadb<10.7")
+    def test_uuid_char(self):
+        self.assert_compile(sqltypes.Uuid(), "CHAR(32)")
+
+    @testing.combinations(
+        (sqltypes.Uuid(), (10, 6, 5), True, "CHAR(32)"),
+        (sqltypes.Uuid(), (10, 6, 5), False, "CHAR(32)"),
+        (sqltypes.Uuid(), (10, 7, 0), True, "UUID"),
+        (sqltypes.Uuid(), (10, 7, 0), False, "CHAR(32)"),
+        (sqltypes.UUID(), (10, 6, 5), True, "UUID"),
+        (sqltypes.UUID(), (10, 6, 5), False, "UUID"),
+        (sqltypes.UUID(), (10, 7, 0), True, "UUID"),
+        (sqltypes.UUID(), (10, 7, 0), False, "UUID"),
+    )
+    def test_uuid_combinations(self, type_, version, native_uuid, res):
+        dialect = MariaDBDialect()
+        dialect.server_version_info = version
+        type_.native_uuid = native_uuid
+        self.assert_compile(type_, res, dialect=dialect)
+
+
 class TypeRoundTripTest(fixtures.TestBase, AssertsExecutionResults):
     __dialect__ = mysql.dialect()
```

Comment:
**Federico Caselli** (CaselIT) wrote:

the lines are duplicated. Did you want to use the type from the dialect module?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5120

Reply:
On github branch it has additional param  native_uuid in this test
but on gerrit it was removed

Reply:
**Michael Bayer** (zzzeek) wrote:

right because all UUID()s are native_uuid=True, so I moved that param to the `Uuid()` types themselves

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5120